libraries/common/src/main/java/androidx/media3/common/util/CodecSpecificDataUtil.java

```diff
@@ -72,6 +78,83 @@ public static boolean parseCea708InitializationData(List<byte[]> initializationD
         && initializationData.get(0)[0] == 1;
   }
 
+  /**
+   * Parses an MPEG-4 Visual configuration information, as defined in ISO/IEC14496-2
+   *
+   * @param videoSpecificConfig A byte array containing the MPEG-4 Visual configuration information
+   *     to parse.
+   * @return A pair consisting of the width and the height.
+   */
+  public static Pair<Integer, Integer> parseMpeg4VideoSpecificConfig(byte[] videoSpecificConfig) {
+    int offset = 0;
+    boolean foundVOL = false;
+    ParsableByteArray scdScratchBytes = new ParsableByteArray(videoSpecificConfig);
+    while (offset + 3 < videoSpecificConfig.length) {
+      if (scdScratchBytes.readUnsignedInt24() != VISUAL_OBJECT_LAYER
+          || (videoSpecificConfig[offset + 3] & 0xf0) != VISUAL_OBJECT_LAYER_START) {
+        scdScratchBytes.setPosition(scdScratchBytes.getPosition() - 2);
+        offset++;
+        continue;
+      }
+      foundVOL = true;
+      break;
+    }
+
+    Assertions.checkArgument(foundVOL, "Invalid input. VOL not found");
+
+    ParsableBitArray scdScratchBits = new ParsableBitArray(videoSpecificConfig);
+    scdScratchBits.skipBits((offset + 4) * 8);
+    scdScratchBits.skipBits(1); // random_accessible_vol
+    scdScratchBits.skipBits(8); // video_object_type_indication
+
+    if (scdScratchBits.readBit()) { // object_layer_identifier
+      scdScratchBits.skipBits(4); // video_object_layer_verid
+      scdScratchBits.skipBits(3); // video_object_layer_priority
+    }
+
+    int aspectRatioInfo = scdScratchBits.readBits(4);
+    if (aspectRatioInfo == EXTENDED_PAR) {
+      scdScratchBits.skipBits(8); // par_width
+      scdScratchBits.skipBits(8); // par_height
+    }
+
+    if (scdScratchBits.readBit()) { // vol_control_parameters
+      scdScratchBits.skipBits(2); // chroma_format
+      scdScratchBits.skipBits(1); // low_delay
+      if (scdScratchBits.readBit()) { // vbv_parameters
+        scdScratchBits.skipBits(79);
+      }
+    }
+
+    int videoObjectLayerShape = scdScratchBits.readBits(2);
+    Assertions.checkArgument(videoObjectLayerShape == RECTANGULAR, "Unsupported feature");
```

Comment:
Change error message to "Only supports rectangular video object layer shape." or something similar